tests/console-backend-service/internal/domain/serverless/function_test.go

```diff
@@ -0,0 +1,308 @@
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const sleepTime = 5 * time.Second
+
+func TestFunctionEventQueries(t *testing.T) {
+	c, err := graphql.New()
+	assert.NoError(t, err)
+
+	coreCli, _, err := client.NewClientWithConfig()
+	require.NoError(t, err)
+
+	namespace, err := fixNamespace(coreCli)
+	require.NoError(t, err)
+	namespaceName := namespace.Name()
+	defer namespace.Delete()
+
+	subscription := subscribeFunctionEvent(c, createFunctionEventArguments("1", namespaceName), functionEventDetailsFields())
+	defer subscription.Close()
+
+	err = mutationFunction(c, "createFunction", mutationFunctionArguments("1", namespaceName, nil), functionDetailsFields())
+	require.NoError(t, err)
+
+	event, err := readFunctionEvent(subscription)
+	require.NoError(t, err)
+
+	expectedFunction := fixFunction("1", namespaceName, nil)
+	expectedEvent := fixFunctionEvent("ADD", expectedFunction)
+	checkFunctionEvent(t, expectedEvent, event)
+
+	//wait for reactions from function controller to function CR
+	time.Sleep(sleepTime) // happy sleep is happy sleep when he's sleeping :)
+
+	function, err := queryFunction(c, queryFunctionArguments("1", namespaceName), functionDetailsFields())
+	require.NoError(t, err)
+	checkFunctionQuery(t, expectedFunction, function)
+
+	labels := []string{FunctionLabel}
+	err = mutationFunction(c, "updateFunction", mutationFunctionArguments("1", namespaceName, labels), functionDetailsFields())
+	require.NoError(t, err)
+
+	function, err = queryFunction(c, queryFunctionArguments("1", namespaceName), functionDetailsFields())
+	require.NoError(t, err)
+
+	expectedFunction = fixFunction("1", namespaceName, labels)
+	checkFunctionQuery(t, expectedFunction, function)
+
+	err = mutationFunction(c, "deleteFunction", deleteFunctionArguments("1", namespaceName), functionMetadataDetailsFields())
+	require.NoError(t, err)
+
+	err = mutationFunction(c, "createFunction", mutationFunctionArguments("2", namespaceName, labels), functionDetailsFields())
+	require.NoError(t, err)
+	err = mutationFunction(c, "createFunction", mutationFunctionArguments("3", namespaceName, labels), functionDetailsFields())
+	require.NoError(t, err)
+
+	functions, err := queryFunctions(c, namespaceName, functionDetailsFields())
+	names := []string{fmt.Sprintf("%s-%s", FunctionNamePrefix, "2"), fmt.Sprintf("%s-%s", FunctionNamePrefix, "3")}
+	checkFunctionList(t, namespaceName, names, functions)
+
+	functionsMeta := []FunctionMetadataInput{
+		{Name: fmt.Sprintf("%s-%s", FunctionNamePrefix, "2"), Namespace: namespaceName},
+		{Name: fmt.Sprintf("%s-%s", FunctionNamePrefix, "3"), Namespace: namespaceName},
+	}
+	err = mutationFunction(c, "deleteManyFunctions", deleteManyFunctionsArguments(namespaceName, functionsMeta), functionMetadataDetailsFields())
+	require.NoError(t, err)
+
+	functions, err = queryFunctions(c, namespaceName, functionDetailsFields())
+	assert.Equal(t, 0, len(functions))
```

Comment:
Please test graphqlDirectives like here https://github.com/kyma-project/kyma/blob/master/tests/console-backend-service/internal/domain/application/application_test.go#L101 Also please update tests for eventing with testing appropriate directives.

Reply:
By testing directives you are testing user permission in Kyma :)

Reply:
Good idea. Thanks! Done